Header header
uint32[] histogram
